### PR TITLE
fix: Ensure email is sent after finalizing an invoice

### PR DESCRIPTION
**Pull request by rsempe**
We have some issues with not sent emails after updating the grace period to 0.
The goal of this is to fix them by removing a transaction.

### CODE REVIEW

File: spec/scenarios/customers/update_invoice_grace_period_spec.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    subscription = customer.subscriptions.find_by(external_id: customer.external_id)
+    dec_invoice = subscription.invoices.first
+    expect(dec_invoice).to be_draft
+
+    ### 1 Jan: Billing
+    jan1 = DateTime.new(2023, 1, 1)
+
+    travel_to(jan1) do
+      expect(InvoiceMailer)
```
Comment (jdenquin) [inline spec/scenarios/customers/update_invoice_grace_period_spec.rb:43]: RSpec/VoidExpect: Do not use `expect()` without `.to` or `.not_to`. Chain the methods or remove it. (https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/VoidExpect)

File: spec/scenarios/customers/update_invoice_grace_period_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+      expect do
+        create_or_update_customer(
+          {
+            external_id: customer.external_id,
+            billing_configuration: { invoice_grace_period: 0 },
+          }
```
Comment (jdenquin) [inline spec/scenarios/customers/update_invoice_grace_period_spec.rb:50]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

**Review by vincent-pochet — APPROVED** [review]
:rocket:

File: app/services/invoices/finalize_service.rb [hunk 3]
```diff
@@ -14,13 +14,16 @@ def call
...
+        invoice.issuing_date = issuing_date
+        invoice.payment_due_date = payment_due_date
+        invoice.save!
 
         invoice.credit_notes.each(&:finalized!)
       end
 
       SendWebhookJob.perform_later('invoice.created', result.invoice) if invoice.organization.webhook_endpoints.any?
-      InvoiceMailer.with(invoice:).finalized.deliver_later if should_deliver_email?
+      InvoiceMailer.with(invoice: invoice).finalized.deliver_later if should_deliver_email?
```
Comment (jdenquin) [inline app/services/invoices/finalize_service.rb:26]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

**Pull request merged**